libraries/Microsoft.Bot.Builder.Skills/ChannelApiMiddleware.cs

```diff
@@ -0,0 +1,138 @@
+                            invokeArgs.Result = new ResourceResponse(id: Guid.NewGuid().ToString("N"));
+                            return;
+                        }
+
+                        invokeArgs.Result = await turnContext.SendActivityAsync(activityPayload, cancellationToken).ConfigureAwait(false);
+                        return;
+                    }
+
+                // Send activity(replyToId, activity)
+                case ChannelApiMethods.ReplyToActivity:
+                    {
+                        var activityPayload = (Activity)invokeArgs.Args[1];
+                        activityPayload.ReplyToId = (string)invokeArgs.Args[0];
+
+                        if (activityPayload.Type == ActivityTypes.EndOfConversation)
+                        {
+                            await ProcessEndOfConversationAsync(turnContext, next, activityPayload, cancellationToken).ConfigureAwait(false);
+                            invokeArgs.Result = new ResourceResponse(id: Guid.NewGuid().ToString("N"));
+                            return;
+                        }
+
+                        invokeArgs.Result = await turnContext.SendActivityAsync(activityPayload, cancellationToken).ConfigureAwait(false);
+                        return;
+                    }
+
+                // UpdateActivity(activity)
+                case ChannelApiMethods.UpdateActivity:
+                    invokeArgs.Result = await turnContext.UpdateActivityAsync((Activity)invokeArgs.Args[0], cancellationToken).ConfigureAwait(false);
+                    return;
+
+                // DeleteActivity(activityId)
+                case ChannelApiMethods.DeleteActivity:
+                    await turnContext.DeleteActivityAsync((string)invokeArgs.Args[0], cancellationToken).ConfigureAwait(false);
+                    break;
+
+                // SendConversationHistory(history)
+                case ChannelApiMethods.SendConversationHistory:
+                    invokeArgs.Result = await turnContext.Adapter.SendConversationHistoryAsync(turnContext, (Transcript)invokeArgs.Args[0]).ConfigureAwait(false);
+                    break;
+
+                // GetConversationMembers()
+                case ChannelApiMethods.GetConversationMembers:
+                    invokeArgs.Result = await turnContext.Adapter.GetConversationMembersAsync(turnContext, cancellationToken).ConfigureAwait(false);
+                    break;
+
+                // GetConversationPageMembers((int)pageSize, continuationToken)
+                case ChannelApiMethods.GetConversationPagedMembers:
+                    invokeArgs.Result = await turnContext.Adapter.GetConversationPagedMembersAsync(turnContext, (int)invokeArgs.Args[0], (string)invokeArgs.Args[1], cancellationToken).ConfigureAwait(false);
+                    break;
+
+                // DeleteConversationMember(memberId)
+                case ChannelApiMethods.DeleteConversationMember:
```

Comment:
>case ChannelApiMethods.DeleteConversationMember: [](start = 16, length = 48)

Another questionable function to expose to skills

Reply:
Same as above, we should probably comment it out for now.